src/m365/spo/commands/file/file-list.ts

```diff
@@ -26,42 +29,67 @@ class SpoFileListCommand extends SpoCommand {
     }
 
-    let requestUrl: string = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${encodeURIComponent(args.options.folder)}')/Files`;
 
-    if (args.options.output !== 'json') {
-      requestUrl += '?$select=UniqueId,Name,ServerRelativeUrl';
-    }
+    // Recursive function
+    const getFiles = (folderUrl: string, files: FilePropertiesCollection={value:[]}):Promise<FilePropertiesCollection> => {
 
-    const requestOptions: any = {
-      url: requestUrl,
-      method: 'GET',
-      headers: {
-        'accept': 'application/json;odata=nometadata'
-      },
-      responseType: 'json'
+      // If --recursive option is specified, retrieve both Files and Folder details, otherwise only Files.
+      const expandParameters:string = args.options.recursive ? 'Files,Folders':'Files';
+      let requestUrl = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${encodeURIComponent(folderUrl)}')?$expand=${expandParameters}`;
+      if (args.options.output !== 'json') {
+        requestUrl += '&$select=Files/UniqueId,Files/Name,Files/ServerRelativeUrl';
+      }
+      const requestOptions:any = {
+        url: requestUrl,
+        method: 'GET',
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+      
+      return request.get<FileFolderCollection>(requestOptions)
+        .then((filesAndFoldersResult: FileFolderCollection) => {
+          return Promise.all(filesAndFoldersResult.Files.map((file:FileProperties) => files.value.push(file)))
+            .then(() => {
+              // If the request is --recursive, call this method for other folders.
+              if(args.options.recursive && filesAndFoldersResult.Folders !== undefined && filesAndFoldersResult.Folders.length !== 0){
+                return Promise.all(filesAndFoldersResult.Folders.map((folder: { ServerRelativeUrl: string; }) => getFiles(folder.ServerRelativeUrl, files)));
+              }
+              else{
+                return;
+              }
+            });
+        }).then(() => files);
     };
 
-    request
-      .get<FilePropertiesCollection>(requestOptions)
-      .then((fileProperties: FilePropertiesCollection): void => {
-        logger.log(fileProperties.value);
+    getFiles(args.options.folder).then((files: FilePropertiesCollection): void => {
+      logger.log(files.value);      
 
-        cb();
-      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+      cb();
+    }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    
   }
 
+
   public options(): CommandOption[] {
     const options: CommandOption[] = [
       {
         option: '-u, --webUrl <webUrl>'
       },
       {
         option: '-f, --folder <folder>'
+      },
+      {
+        option: '-r, --recursive [recursive]'
```

Comment:
Like mentioned in the docs, let's remove `[recursive]` so that it's clear that it's meant to be used as a flag rather than an option with a value.